etlas-cabal/Distribution/Parsec/Class.hs

```diff
@@ -125,12 +126,11 @@ instance Parsec ModuleName where
         validModuleChar c = isAlphaNum c || c == '_' || c == '\''
 
 instance Parsec FlagName where
-    parsec = mkFlagName . map toLower . intercalate "-" <$> P.sepBy1 component (P.char '-')
+    parsec = mkFlagName . lowercase <$> parsec'
       where
-        -- http://hackage.haskell.org/package/cabal-debian-4.24.8/cabal-debian.cabal
-        -- has flag with all digit component: pretty-112
-        component :: P.Stream s Identity Char => P.Parsec s [PWarning] String
-        component = P.munch1 (\c -> isAlphaNum c || c `elem` "_")
+        parsec' = (:) <$> lead <*> rest
+        lead = P.satisfy (\c ->  isAlphaNum c || c == '_')
+        rest = P.munch (\c -> isAlphaNum c || c == '_' || c == '-')
```

Comment:
Ok looks like a refactor.

Reply:
It is weird but testing with cabal files gave me an error when indexing cassava with PARSEC option. It has an infamous flag with an internal double dash. Dont know why previous etlas version that has PARSEC enabled didint throwed the error
(See https://github.com/typelead/etlas/pull/72/commits/184700573eb1a4048f3be796b894abdffbe3d451)

Reply:
Ohhh, ok, however i think i've used etlas installed from its own repo and i dont remember having that error (cassava is always indexed?).. anyway afaik the new flag parser is more robust cause it will support previous and future flag formats

Reply:
Great work! :)